### PR TITLE
chore: [IA-867] Add monitoring events for the Poste Data Matrix

### PR DESCRIPTION
**Pull request by dgopsq**
## Short description
This PR is going to add two events used to track the Poste Data Matrix barcode scan feature.

✅ Legal compliance: https://pagopa.atlassian.net/browse/IA-869

## List of changes proposed in this pull request
- Added a `WALLET_SCAN_POSTE_DATAMATRIX_FAILURE` if a failure happened while scanning the barcode
- Added a `WALLET_SCAN_POSTE_DATAMATRIX_SUCCESS` if a Data Matrix barcode from Poste has been scanned successfully

## How to test
Scanning a Data Matrix should trigger these events through the `mixpanelTrack` function.

<details>
<summary>Valid Data Matrix example</summary>

valid

</details>

<details>
<summary>Invalid Data Matrix example</summary>

Invalid

</details>

### CODE REVIEW

File: ts/screens/wallet/payment/ScanQrCodeScreen.tsx [hunk 1]
```diff
@@ -184,8 +185,17 @@ class ScanQrCodeScreen extends React.Component<Props, State> {
       const maybePosteDataMatrix = decodePosteDataMatrix(data);
 
       return maybePosteDataMatrix.foldL<void>(
-        this.onInvalidQrCode,
-        this.onValidQrCode
+        () => {
+          if (this.state.scanningState !== "INVALID") {
+            void mixpanelTrack("WALLET_SCAN_POSTE_DATAMATRIX_FAILURE");
```
Comment (fabriziofff) [inline ts/screens/wallet/payment/ScanQrCodeScreen.tsx:190]: Could you add to https://docs.google.com/spreadsheets/d/1bXNRcTDKU-5l2RqCcwSbT7Y1M9QQTWtS_f-3-AIqSYk/edit#gid=0 these new two events?

**Review by fabriziofff — CHANGES_REQUESTED** [review]

File: ts/screens/wallet/payment/ScanQrCodeScreen.tsx (hunk 1 above)
Comment (dgopsq) [reply ts/screens/wallet/payment/ScanQrCodeScreen.tsx:190]: @fabriziofff can you give me writing permissions on that file?

Comment (fabriziofff) [reply ts/screens/wallet/payment/ScanQrCodeScreen.tsx:190]: Ops, you should have all the permissions now!

Comment (dgopsq) [reply ts/screens/wallet/payment/ScanQrCodeScreen.tsx:190]: @fabriziofff Done ✅

**Review by fabriziofff — APPROVED** [review]

**Pull request merged**